src/libcprover-cpp/options.h

```diff
@@ -0,0 +1,26 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#ifndef CPROVER_LIBCPROVER_CPP_OPTIONS_H
+#define CPROVER_LIBCPROVER_CPP_OPTIONS_H
+
+#include <memory>
+
+class optionst;
+
+class api_optionst
+{
+  // Options for the verification engine
+  bool simplify_enabled;
+
+  // Private interface methods
+  api_optionst() = default;
+
+public:
+  static api_optionst create();
+
+  api_optionst &simplify(bool on);
+
+  std::unique_ptr<optionst> to_engine_options() const;
```

Comment:
These strictly need comments. This is supposed to be an API to be used, after all.